src/main/java/uk/gov/ch/service/officer/active/impl/ActiveOfficerDetailsServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.ch.service.officer.active.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import uk.gov.ch.OracleQueryApplication;
+import uk.gov.ch.exception.InvalidActiveOfficersCountFoundException;
+import uk.gov.ch.model.officer.active.ActiveOfficerDetails;
+import uk.gov.ch.repository.officers.ActiveOfficersDetailsRepository;
+import uk.gov.ch.service.officer.active.ActiveOfficerDetailsService;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.List;
+
+@Service
+public class ActiveOfficerDetailsServiceImpl implements ActiveOfficerDetailsService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OracleQueryApplication.APPLICATION_NAME_SPACE);
+
+    private static final String WRONG_OFFICER_COUNT_MESSAGE = "Incorrect number of officers found must be single active director company.";
+    private static final String NO_OFFICERS_FOUND_MESSAGE = "No active officers were found.";
+
+    @Autowired
+    private ActiveOfficersDetailsRepository activeOfficersDetailsRepository;
+
+    @Override
+    public ActiveOfficerDetails getActiveDirectorDetails(String companyNumber, Pageable pageable) throws InvalidActiveOfficersCountFoundException {
+        List<ActiveOfficerDetails> list =  activeOfficersDetailsRepository.getActiveDirectorDetails(companyNumber, pageable).toList();
+        if (list.size() == 1 ) {
+            return list.get(0);
+        } else {
+            LOGGER.error(WRONG_OFFICER_COUNT_MESSAGE + companyNumber);
+            throw new InvalidActiveOfficersCountFoundException(WRONG_OFFICER_COUNT_MESSAGE);
+        }
+    }
+
+    @Override
+    public List<ActiveOfficerDetails> getActiveOfficersDetails(String companyNumber, Pageable pageable) throws InvalidActiveOfficersCountFoundException {
+        List<ActiveOfficerDetails> list =  activeOfficersDetailsRepository.getActiveOfficersDetails(companyNumber, pageable).toList();
+        LOGGER.debug(String.format("Query list size returned = %s", list.size()));
```

Comment:
can't remember if the debug will log the method name but just in case, add the name of the call into the debug message, otherwise logs might just say Query list size returned = 5 and we won't know what it relates to. So have something like ' getActiveOfficersDetails Query list size returned = %s'